module/mempool/stdmap/identifier_map.go

```diff
@@ -70,18 +58,13 @@ func (i *IdentifierMap) Append(key, id flow.Identifier) error {
 // Otherwise it returns nil and false.
 func (i *IdentifierMap) Get(key flow.Identifier) ([]flow.Identifier, bool) {
 	ids := make([]flow.Identifier, 0)
-	err := i.Run(func(backdata mempool.BackData) error {
-		entity, ok := backdata.ByID(key)
+	err := i.Run(func(backdata mempool.BackData[flow.Identifier, map[flow.Identifier]struct{}]) error {
```

Comment:
```suggestion
	// we need to perform a blocking operation since we are dealing with a reference object. Since our mempool
	// changes the map itself in other operations we need to ensure that all changes to the map are done in mutually
	// exclusive way, otherwise we risk observing a race condition. This is exactly why we perform `Get` and 
	// transformation  of the map in critical section since if the goroutine will be suspended in between those two operations
	// we are potentially concurrently accessing the map.
	err := i.Run(func(backdata mempool.BackData[flow.Identifier, map[flow.Identifier]struct{}]) error {
```

This is very complicated, I've spent a few minutes to understand why we need a `Run` here.